src/Xamarin.Android.Tools.Bytecode/ClassPath.cs

```diff
@@ -85,6 +82,22 @@ public void Load (Stream jarStream, bool leaveOpen = false)
 			}
 		}
 
+		static bool ShouldLoadEntry (ZipArchiveEntry entry)
+		{
+			if (entry.Length == 0)
+				return false;
+
+			if (entry.Name == "module-info.class")
+				return false;
+
+			if (entry.Name.EndsWith (".jnilib", StringComparison.OrdinalIgnoreCase))
+				return false;
+
+			using var s = entry.Open ();
```

Comment:
In the context of #1092, should this instead use `BufferedStream`?

Reply:
I don't think we want it in this case, as we are reading a single value from the `Stream` and then closing it.  If we use `BufferedStream` it would need to read xKB into the buffer instead of the single 4 byte read.